lib/LedgerSMB/Template.pm

```diff
@@ -270,26 +282,104 @@ sub _valid_language {
 }
 
 sub _preprocess {
-    my ($self, $vars) = @_;
-    return unless $self->{myconfig};
-    use LedgerSMB;
-    my $type = ref($vars);
+    my ($rawvars, $escape) = @_;
+    return undef unless defined $rawvars;
 
-    if ($type eq 'SCALAR' || !$type){
-        return;
+    local ($@);
+    if (eval {$rawvars->can('to_output')}){
+        $rawvars = $rawvars->to_output;
     }
-    if ($type eq 'ARRAY'){
-        for (@$vars){
-            if (ref($_)){
-                $self->_preprocess($_);
-            }
+    use LedgerSMB;
```

Comment:
Hmm. We'd want this use statement at the top of the file.

But more importantly, what do we need the LedgerSMB module for specifically? I'm not seeing it being used below...